### PR TITLE
[feature] Enable sentry logging for APIv2

### PR DESCRIPTION
**Pull request by abought**
Refs #3163 

## Purpose
Allow Sentry to track errors that happen in the Django app used by APIv2.

## Summary of changes
Adds raven middleware to Django. Raven is already listed as a requirement for flask, so no new user installation is needed.

Applies a fix to ensure that Sentry catches problems when used with other error handling middleware, as [suggested](http://raven.readthedocs.org/en/latest/integrations/django.html#error-handling-middleware) by the Sentry docs.

## Blockers
Not sure how to test this locally, since I don't have the SENTRY_DSN keys needed to send actual data to the server.

Feedback on more granular reporting options would also be welcome. It would be nice if we could filter API/django errors out from the OSF/flask errors in a clear way. @brianjgeiger ?

### CODE REVIEW

File: api/base/middleware.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
-from framework.transactions import commands, messages, utils
 from pymongo.errors import OperationFailure
+from raven.contrib.django.raven_compat.models import sentry_exception_handler
+from framework.transactions import commands, messages, utils
```
Comment (lyndsysimon) [inline api/base/middleware.py]: Needs a blank line before the `framework` import.

ref: http://cosdev.readthedocs.org/en/latest/style_guides/python.html#imports

**Comment by lyndsysimon** [comment]
First pass complete. Only one minor change requested, ready for merge now if you'd like. :last_quarter_moon:

**Comment by abought** [comment]
Response to review pushed. 

Then learned about [ci skip], which I should totally have used. :)

File: api/base/middleware.py [hunk 2]
```diff
@@ -19,6 +21,7 @@ def process_exception(self, request, exception):
         """If an exception occurs, rollback the current transaction
         if it exists.
         """
+        sentry_exception_handler(request=request)
```
Comment (sloria) [inline api/base/middleware.py]: Will this cause an error if Sentry is not used in local development?

Comment (abought) [inline api/base/middleware.py]: Raven does its own checks to ensure that the client is active. captureException (used internally) returns None is no dsn key is provided, and when the API app is loaded, the message `Raven is not configured (logging is disabled). Please see the documentation for more information.` appears on the screen.

**Comment by abought** [comment]
@sloria , any progress on reviewing this one?

**Comment by sloria** [comment]
Looks good. :octopus:

**Pull request merged**